src/main/java/org/graylog/integrations/aws/AWSClientBuilderUtil.java

```diff
@@ -0,0 +1,100 @@
+package org.graylog.integrations.aws;
+
+import com.google.common.base.Preconditions;
+import org.apache.commons.lang3.StringUtils;
+import org.graylog.integrations.aws.resources.requests.AWSRequest;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.awscore.client.builder.AwsClientBuilder;
+import software.amazon.awssdk.core.client.builder.SdkClientBuilder;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.cloudwatchlogs.CloudWatchLogsClient;
+import software.amazon.awssdk.services.cloudwatchlogs.CloudWatchLogsClientBuilder;
+import software.amazon.awssdk.services.iam.IamClient;
+import software.amazon.awssdk.services.iam.IamClientBuilder;
+import software.amazon.awssdk.services.kinesis.KinesisClient;
```

Comment:
https://docs.aws.amazon.com/sdk-for-java/v2/developer-guide/examples-kinesis-stream.html
When do we use KinesisAsyncClient vs KinesisClient ?

Reply:
@lingpri Great question. The synchronous Kinesis client is used in most of the API calls in [KinesisService](https://github.com/Graylog2/graylog-plugin-integrations/blob/91e8e1190b1e6c460a462977e9451b4451d71dbb/src/main/java/org/graylog/integrations/aws/service/KinesisService.java). We decided to use the sync client for all API requests here, because there these requests are blocking in the UI anyhow. The async client is used in [KinesisConsumer](https://github.com/Graylog2/graylog-plugin-integrations/blob/eb48aad6c9c1a90ba4c8832fe7a082162a0848ee/src/main/java/org/graylog/integrations/aws/transports/KinesisConsumer.java#L40) for reading messages from KinesisStreams in the normal runtime of the input. This usage requires async.

Reply:
@danotorrey - Bingo . I was thinking about how to make the UI Non-blocking yesterday. I'm wondering if we can bring the **CLI/Terminal** in a wizard like interface.